waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/FederatedHMSHandler.java

```diff
@@ -1369,19 +1372,48 @@ public List<String> set_ugi(String user_name, List<String> group_names) throws M
   @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
   public String get_delegation_token(String token_owner, String renewer_kerberos_principal_name)
       throws MetaException, TException {
-    return getPrimaryClient().get_delegation_token(token_owner, renewer_kerberos_principal_name);
+    try {
+      return saslServerWrapper.getDelegationTokenManager()
+          .getDelegationToken(token_owner, renewer_kerberos_principal_name,
+              saslServerWrapper.getIPAddress());
+    } catch (IOException | InterruptedException e) {
+      throw new MetaException(e.getMessage());
+    }
   }
 
   @Override
   @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
   public long renew_delegation_token(String token_str_form) throws MetaException, TException {
-    return getPrimaryClient().renew_delegation_token(token_str_form);
+    try {
+      return saslServerWrapper.getDelegationTokenManager()
+          .renewDelegationToken(token_str_form);
+    } catch (IOException e) {
+      throw new MetaException(e.getMessage());
+    } catch (Exception e) {
+      throw newMetaException(e);
+    }
   }
 
   @Override
   @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
   public void cancel_delegation_token(String token_str_form) throws MetaException, TException {
-    getPrimaryClient().cancel_delegation_token(token_str_form);
+    try {
+      saslServerWrapper.getDelegationTokenManager()
+          .cancelDelegationToken(token_str_form);
+    } catch (IOException e) {
```

Comment:
Both exceptions are wrapped in a `MetaException`. Why not capture all exception types in a single catch statement and wrap them there?

Reply:
Actually, I also find it strange, but Hive write it this way, and my writing is mainly consistent with Hive. [https://github.com/apache/hive/blob/branch-3.1/standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java#L6596](url)

Reply:
Then let's leave it like that. @patduin told me that a lot of this code is coming from Hive, so we don't need to change it. I'll resolve this comment.